packages/@aws-cdk/aws-synthetics-alpha/lib/canary.ts

```diff
@@ -10,6 +10,11 @@ import { Runtime } from './runtime';
 import { Schedule } from './schedule';
 import { CloudWatchSyntheticsMetrics } from 'aws-cdk-lib/aws-synthetics/lib/synthetics-canned-metrics.generated';
 import { CfnCanary } from 'aws-cdk-lib/aws-synthetics';
+import { CustomResource, CustomResourceProvider, CustomResourceProviderRuntime } from 'aws-cdk-lib/core';
+import * as path from 'path';
+
+const AUTO_DELETE_UNDERLYING_RESOURCES_RESOURCE_TYPE = 'Custom::SyntheticsAutoDeleteUnderlyingResources';
```

Comment:
i'm sure you're going to hate how long/awkward this name is but I don't have a better one

Reply:
Fortunately it's private so I don't have to care 😉 .

I might have gone for something with "cleanup" in the name, that is sorta a synonym to "auto delete underlying resources".